streams/src/test/java/org/apache/kafka/streams/integration/utils/EmbeddedKafkaCluster.java

```diff
@@ -284,22 +283,44 @@ public void deleteAndRecreateTopics(final long timeoutMs, final String... topics
         createTopics(topics);
     }
 
+    public void waitForRemainingTopics(final long timeoutMs, final String... topics) throws InterruptedException {
```

Comment:
IMHO seems like the test conditions don't belong in `EmbeddedKafkaCluster`, but I don't have a better solution ATM.

Reply:
I agree. I we want this to be general purpose, we should add this to some test util class.

Reply:
For `EmbeddedKafkaCluster` we should just add a `boolean topicExists(String topic)` method ?

Reply:
I have thought about this, and considered a few different places to consolidate such functions, e.g. in `o.a.k.test.TestUtils`. But the problem is that `kafka.utils.ZkUtils` is needed for this check and hence need to be depend on that package. That's why I ended up in this situation (admittedly not ideal). If there is a better suggestion please let me know.

Regarding `boolean topicExists(String topic)`, the reason I cannot just do that is 1) it needs to have a timeout since we need to wait for the ZK path to be deleted, and 2) for this specific test we need to test exact equality.

Reply:
I cannot follow your argument. We need `ZkUtils` to check if a topic exists. Thus, this can be done with `EmbeddedKafkaCluster#topicExists()`. Furthermore, we add a helper method to `TestUils` that also accept a timeout and calls `#topicExists()` in a loop to make sure we get the exact topics we want. Why do you think this is not feasible?